.github/workflows/release.yml

```
@@ -0,0 +1,81 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - r[0-9]+
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  create-release:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v5
+        - uses: ncipollo/release-action@v1
+          env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          with:
+            draft: true
+            allowUpdates: true
+            omitBodyDuringUpdate: true
+
+  build-release:
+    name: Build release binaries
+    runs-on: windows-2022
+    steps:
+      - name: Check out files
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 0
```

```suggestion
          fetch-depth: 1
```

We don't need to fetch everything for release, so we can set the depth to 1

---

> Doesn't this break when i do this?

I don't think so (famous last words), since it release on tag it should in theory always grab the tagged commit anyways

---

Okay well, the thing is that its not based on the latest tag, i was wrong on that.
It does `git rev-list --count HEAD` (`getRevisionNumber()` in premake5.lua).

So it _will_ break because it counts commits...

---

ah well, lets leave it as is